mis_builder/models/mis_report_instance.py

```diff
@@ -133,6 +133,19 @@ def _compute_dates(self):
                 record.date_from = fields.Date.to_string(date_from)
                 record.date_to = fields.Date.to_string(date_to)
                 record.valid = True
+
+            elif record.mode == MODE_REL and record.type == "q":  # Quarterly period logic
+                date_from = d.replace(day=1)
```

Comment:
This always takes the `date_from` as the start of the month, but in case of quarterly logic, it should be the start of the quarter that the `d` date falls in.